jax/interpreters/invertible_ad.py

```diff
@@ -43,24 +42,25 @@
 invertible_call_p.def_impl(core.call_impl)
 invertible_call_p.multiple_results = True
 
-def _invertible_call_make_output_tracers(trace, typed_jaxpr, in_tracers, out_known_pvals, out_unknown_pvals, _, params):
-  unknown_output_tracers = [JaxprTracer(trace, out_pval, None) for out_pval in out_unknown_pvals]
-  lifted_jaxpr = convert_constvars_jaxpr(typed_jaxpr.jaxpr)
-  # Add dummy arguments representing the outputs to the jaxpr. Those should remain unused in case
-  # the expression actually ends up being evaluated, but they make it well-formed.
-  out_known_avals = tuple(raise_to_shaped(get_aval(pval.get_known())) for pval in out_known_pvals)
-  lifted_jaxpr = _append_invars(lifted_jaxpr, out_known_avals)
-  new_params = dict(params, call_jaxpr=lifted_jaxpr)
-  # We also append some dummy outputs that correspond to the known outputs we left in the call_jaxpr
-  dummy_outputs = [JaxprTracer(trace, pval, core.unit) for pval in out_known_pvals]
-
-  output_constants = [JaxprTracer(trace, pval, ConstVar(pval.get_known())) for pval in out_known_pvals]
-  eqn = new_eqn_recipe(tuple(it.chain(in_tracers, output_constants)),
-                       dummy_outputs + unknown_output_tracers,
-                       invertible_call_p,
-                       new_params)
-  for t in unknown_output_tracers: t.recipe = eqn
-  return unknown_output_tracers
+def _invertible_call_make_output_tracers(trace, in_tracers, out_tracers, params):
+  uks = [not t.pval.is_known() for t in out_tracers]
+  out_tracers_known, out_tracers_unknown = _partition_knowns(out_tracers, uks)
+
+  # Add dummy arguments representing the outputs to the jaxpr. Those should
+  # remain unused if the expression is evaluated, but they make it well-formed.
+  out_known_avals = [raise_to_shaped(t.pval.get_aval()) for t in out_tracers_known]
+  out_consts = [trace.instantiate_const(t) for t in out_tracers_known]
+  new_jaxpr = _append_invars(params['call_jaxpr'], tuple(out_known_avals))
+  new_in_tracers = (*in_tracers, *out_consts)
+
+  # Append dummy outputs that correspond to known outputs left in the call_jaxpr
+  dummy_outputs = [JaxprTracer(trace, t.pval, core.unit) for t in out_tracers_known]
```

Comment:
I guess this could have been `trace.new_const(t.pval.get_known())`?

Reply:
I'll give that a shot!

Reply:
Worked like a charm!